ui/src/app/pages/homepage/homepage.tsx

```diff
@@ -156,6 +164,25 @@ export const Homepage = withUserProfile()(class extends React.Component<Props, S
     }
   }
 
+  async validateRasLoginGovLink() {
+    const authCode = (new URL(window.location.href)).searchParams.get('code');
+    const redirectUrl = buildRasRedirectUrl();
+    if (authCode) {
+      this.setState({rasLoginGovLoading: true});
```

Comment:
Is this state currently being used anywhere?

Reply:
that state controls the button status, (search for `loadingPropsKey` in registiration.tsx)